### PR TITLE
Follow up #1670: Fix StyleCop issues and improve the code

### PR DESCRIPTION
**Pull request by raman-m**
## Relates to #1670 
This is follow up PR after merging of PR #1670
- #1670 

## Proposed Changes
- Reviewed the code of the related PR
- Removed code smells
- Fixed StyleCop rules violations
- Made code improvements

### CODE REVIEW

**Comment by raman-m** [comment]
@RaynaldM Hey, Ray!
I will appreciate your review and approval **ASAP**, during this weekend, Saturday or Sunday, Sep 30 or Oct 1.
Thank you!

**Comment by ggnaegi** [comment]
> @ggnaegi FYI
> As we agreed on [PR 1670 merging](1670#issuecomment-1740865731) I've created this PR.

I could review it...

**Comment by raman-m** [comment]
@ggnaegi Your approval will have gray status which means it does not approve the PR actually and uncover PR status to merge.
But your opinion is welcomed! I could even fix something if you judge my code 😉

**Comment by raman-m** [comment]
@ggnaegi Will you review?

**Comment by ggnaegi** [comment]
> @ggnaegi Will you review?

@raman, yes sure, probably in the afternoon...

File: src/Ocelot.Provider.Consul/Consul.cs
Comment (ggnaegi) [inline src/Ocelot.Provider.Consul/Consul.cs:22]: Do we really want to expose this string, it's imho, error prone?  Maybe it would be wise to have a factory as a static method?
```c#
public static string GetServiceValidationWarning(string address, int port, string serviceName)
    {
        return $"Unable to use service address: '{address}' and port: {port} as it is invalid for the service: '{serviceName}'. Address must contain host only e.g. 'localhost', and port must be greater than 0.";
    }
```

**Review by ggnaegi — CHANGES_REQUESTED** [review]
@raman-m I reviewed the code:
- 1) ServiceValidationWarningFormat -> I would use a static factory method, otherwise it's error prone
- 2) ThenTheLoggerHasBeenCalledCorrectlyWithValidationWarning -> I would only one theory instead of 3 facts, otherwise we would have to verify the port or url explicitely to match the fact name
- 3) My bad, The theory and  facts names are wrong -> should_return_provider_according_to_service_name

File: src/Ocelot.Provider.Consul/Consul.cs
Comment (raman-m) [reply src/Ocelot.Provider.Consul/Consul.cs:22]: What is the goal of such wrapping if this method will have one reference in the class where it is used?
Definition of new method makes sense if there are 2+ references.
The message is reused by one user scenario. This scenario is covered by a unit test where we check actual value.

Second, if we define a public method, we have to unit test it!
I'm not a fan to write one additional separated unit test.

Comment (ggnaegi) [reply src/Ocelot.Provider.Consul/Consul.cs:22]: Yeah, but it is the same problem if you are using a public string and reuse it later in unit tests... eg.
```string.Format(ConsulProvider.ServiceValidationWarningFormat, service.Address, service.Port, service.Service);```

Comment (ggnaegi) [reply src/Ocelot.Provider.Consul/Consul.cs:22]: But at the end, it doesn't really matter. I just don't like the fact of having a string with parameters as public variable.

**Review by RaynaldM — DISMISSED** [review]

File: src/Ocelot.Provider.Consul/ConsulRegistryConfiguration.cs
Comment (raman-m) [reply src/Ocelot.Provider.Consul/ConsulRegistryConfiguration.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/679c4fff46d90139b97422f8c1db03df5c71d40a

File: src/Ocelot.Provider.Consul/UnableToSetConfigInConsulError.cs
Comment (raman-m) [reply src/Ocelot.Provider.Consul/UnableToSetConfigInConsulError.cs:9]: Not an issue

File: src/Ocelot.Provider.Eureka/EurekaProviderFactory.cs
Comment (raman-m) [reply src/Ocelot.Provider.Eureka/EurekaProviderFactory.cs:20]: But we've discussed this line in your previous PR :))
OK, going to fix...

**Review by RaynaldM — DISMISSED** [review]

**Comment by raman-m** [comment]
@RaynaldM  Sorry, there will be a number of code review commits... Wait for my next message please!

File: src/Ocelot.Provider.Eureka/EurekaProviderFactory.cs
Comment (raman-m) [reply src/Ocelot.Provider.Eureka/EurekaProviderFactory.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/ef754e101d3334b448a1f257305c7fc45171aa08 and https://github.com/ThreeMammals/Ocelot/pull/1711/commits/39fd5382d28fed08c30194c8aa81298b39d23125

**Review by RaynaldM — DISMISSED** [review]

File: test/Ocelot.UnitTests/Consul/ConsulServiceDiscoveryProviderTests.cs
Comment (raman-m) [reply test/Ocelot.UnitTests/Consul/ConsulServiceDiscoveryProviderTests.cs:196]: A-ha-ha! It seems I found the bug made by Tom in https://github.com/ThreeMammals/Ocelot/commit/11a2d13f180a0444d9cb04fd95f87be6489e09d3 on line [61](../commit/11a2d13f180a0444d9cb04fd95f87be6489e09d3#diff-91a3279f52853dbd72d34a3a759a8438e3e83c77cbbde3ef2c526335ee0e98d7R61)...
If address contains `http` **OR** address contains `https` then it is actually **valid** but it returns `false` 🤣 

SO, assumption in the `should_not_return_services_with_invalid_address` test is wrong because `Address = "http://localhost"` is valid, but the test considers it as invalid because of the bug in boolean formula of the predicate. 🥸 
Okay... Tom... Good job! 🤣

Comment (raman-m) [reply test/Ocelot.UnitTests/Consul/ConsulServiceDiscoveryProviderTests.cs:141]: @ggnaegi Partially fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/5851300815762cfe23304fc23defef2fbbabd119

Regarding `Theory`... You will be able to refactor these 3 tests as you wish in the next PRs... 😉 
Currently I cannot merge 3 tests to one theory because of closure of `_serviceName` private member while `AgentService` creating, but `MemberData` attribute of `Theory` requires only static property/member and it has no access to object members! So, it will be a little challenge for you. 😄

And version you've reviewed just a refactoring, I test the same for which these tests were written. So, port & service name checking is encapsulated inside of the warning expected string.
So, writing something like `actualWarning.ShouldContain(portString)` will be not right.
Finally, I believe, it is not an issue.

Comment (ggnaegi) [reply test/Ocelot.UnitTests/Consul/ConsulServiceDiscoveryProviderTests.cs:141]: @raman-m Ok 😄

File: test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs
Comment (raman-m) [reply test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs:49]: Hmm... BDDfy can work with different name conventions: ["You can use underscored or pascal or camel cased method names for your steps"](https://github.com/TestStack/TestStack.BDDfy/blob/10ee0c95af78828116aac02d7283777a85256ef0/readme.md?plain=1#L7)
But the C# naming convention is PascalCase. As for C# developer, as for me, it is very strange to see C & C++ naming convention with `_` (underscores) in method names!
In future I am going to fix this huge problem.

**2nd** problem. Visual Studio generates a lot of IDE1006 `Naming rule violation` because method name is not PascalCase. 
But for now, it seems scenarios with PascalCase is not logged into build log at all. Probably it is Ok because passed scenario should not be logged by default.
Currently all BDDfy unit tests are logged producing huge logs... But this is another story.

It makes a lot of sense to change names from underscore-style (C++) to C# PascalCase because of the following:
- Passed unit test will not be logged by BDDfy framework to build console on CircleCI
- Method names of unit tests don't violate default [C# naming conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions) (aka IDE1006 rule, etc.)

Comment (ggnaegi) [reply test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs]: @raman-m Ok, Raman it makes sense: So, we could keep it like that and create a new PR for theories / facts naming conventions. I'm not used to the underscore naming convention either.

Comment (raman-m) [reply test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/842a94135973c71c7770070936b2930f9ab16bc5

File: test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs
Comment (raman-m) [reply test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/842a94135973c71c7770070936b2930f9ab16bc5

File: test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs
Comment (raman-m) [reply test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/842a94135973c71c7770070936b2930f9ab16bc5

File: test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs
Comment (raman-m) [reply test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs]: It makes too much sense... But not in this PR which I would like to merge today :)
It will another huge PR to resolve mentioned problems... in October from me... 😸 

It seems I must create another follow up issue after this follow up... 🤣 I'm going crazy with this project...

File: src/Ocelot.Provider.Consul/Consul.cs
Comment (raman-m) [reply src/Ocelot.Provider.Consul/Consul.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/eefa04272e6b8f435408f8eef23132194b7706d7

**Comment by raman-m** [comment]
@ggnaegi I've resolved issues of your code review?
Are we done?

**Comment by raman-m** [comment]
@RaynaldM I'm sorry that I dismissed your approvals multiple times!
Make a final approval, plz 🙏

**Review by ggnaegi — APPROVED** [review]
Yes!

**Review by RaynaldM — APPROVED** [review]

**Comment by raman-m** [comment]
<sub>6aa94b54 by **raman-m** on Tuesday, October 03 at 11:05 &rarr;</sub>
Rename Release Notes file
<sub>e5f31ef0 by **Raman Maksimchuk** on Monday, October 02 at 19:17 &rarr;</sub>
Follow up #1670: Fix StyleCop issues and improve the code (#1711)
<sub>e950cf2f by **Ben Bartholomew** on Saturday, September 30 at 21:50 &rarr;</sub>
#653 Adding null check for downstreamResponse (#1334)
<sub>5a81cce1 by **Freddy** on Saturday, September 30 at 20:34 &rarr;</sub>
Second parameter of Substring is wrong in HeaderFindAndReplaceCreator (#1424)
<sub>5dbbbefc by **Fabrizio Mancin** on Saturday, September 30 at 17:26 &rarr;</sub>
#1400 Manage multiple patterns for allowed/blocked IPs via Security Options config section (#1399)
<sub>cc0b9b83 by **Kevin Grossmann** on Friday, September 29 at 21:42 &rarr;</sub>
Ignore port when using UpstreamHost option (#1505)
<sub>f8eb5e1c by **Guillaume Gnaegi** on Friday, September 29 at 20:44 &rarr;</sub>
#1634 #1487 #1329 #1304 #1294 #793 Consul polling of services: enhancements and fix errors (#1670)
<sub>ab294423 by **Roman** on Friday, September 29 at 11:27 &rarr;</sub>
#1509 #1683 Replace non-WS protocols for the 'ClientWebSocket' in WebSocketsProxyMiddleware (#1689)
<sub>190b0017 by **leonluc-dev** on Thursday, September 28 at 19:14 &rarr;</sub>
#1225 Update ServiceDiscovery documentation and samples to include Custom Providers (#1656)
<sub>fa179bf4 by **zhangq** on Thursday, September 28 at 18:04 &rarr;</sub>
#1375 #1237 #925 #920 Fix DownstreamRoute DangerousAcceptAnyServerCertificateValidator (#1377)
<sub>5fd5bf93 by **Raman Maksimchuk** on Monday, September 25 at 18:00 &rarr;</sub>
#1681 'using' statements life hacks (#1682)
<sub>8b892574 by **Raman Maksimchuk** on Monday, September 25 at 16:54 &rarr;</sub>
#1700 Add sequential tests with disabled parallelization (#1705)
<sub>fdad15d4 by **Guillaume Gnaegi** on Monday, September 25 at 15:15 &rarr;</sub>
#1700 Create custom Ocelot config file when instantiating steps during parallel execution (#1703)
<sub>75cd0b30 by **Raman Maksimchuk** on Monday, September 25 at 10:51 &rarr;</sub>
#1677 #1686 United documentation update (#1678)
<sub>b7976fcc by **guoyongchang** on Saturday, September 23 at 17:30 &rarr;</sub>
#1351 #1510 New AddOcelot extensions to support custom ASP.NET MVC pipeline building via the IMvcCoreBuilder interface (#1655)
<sub>4d245ec7 by **Vijay Karavadra** on Saturday, September 23 at 16:49 &rarr;</sub>
#1228 #1235 #1247 Overloaded AddOcelot method for dynamic FileConfiguration construction (#1569)
<sub>5ad807d7 by **jlukawska** on Saturday, September 23 at 12:12 &rarr;</sub>
#893 Problems to cancel a request (CancellationToken) (#1367)
<sub>b27761f4 by **jlukawska** on Saturday, September 23 at 11:43 &rarr;</sub>
#1057 #1068 Add service discovery logs (#1366)
<sub>ec85b132 by **jlukawska** on Friday, September 22 at 22:39 &rarr;</sub>
#1288 Query string char missing (#1335)
<sub>8893f049 by **Marco** on Friday, September 22 at 22:25 &rarr;</sub>
fix: minor grammar fix in default rate limit response message (#1639)
<sub>e6081cef by **Thiago Loureiro** on Friday, September 22 at 22:15 &rarr;</sub>
Fixed logger message typos in the OutputCacheMiddleware class (#1315)
<sub>21bc4a4a by **DanHarltey** on Friday, September 22 at 22:05 &rarr;</sub>
#908 Quality of Service - Configuration (#1279)
<sub>455d7d3f by **Anthony Steele** on Friday, September 22 at 21:45 &rarr;</sub>
Exception mapper naming (#1211)
<sub>818d1fcd by **Raman Maksimchuk** on Friday, September 22 at 21:29 &rarr;</sub>
Merge pull request #1697 from ThreeMammals/main

**Comment by raman-m** [comment]
**4** :star::star::star::star: to **Raman Maksimchuk** 🥇 
**3** :star::star::star: to **jlukawska** 🥈 
**2** :star::star: to **Guillaume Gnaegi** 🥉 

---

### Honoring :medal_sports: aka Top Contributors :clap:
1st :1st_place_medal: goes to **Raman Maksimchuk** for delivering **4** commits
2nd :2nd_place_medal: goes to **jlukawska** for delivering **3** commits
3rd :3rd_place_medal: goes to **Guillaume Gnaegi** for delivering **2** commits in **2** files changed
3rd :3rd_place_medal: goes to **Ben Bartholomew** for delivering **1** feature in **3** files changed with **83** insertions
3<sup>rd</sup> :3rd_place_medal: goes to **Guillaume Gnaegi** for delivering **1** feature in **3** files changed with **13** insertions and with **422** deletions

### Starring :star: aka Release Influencers :bowtie:
:star::star::star::star:  Raman Maksimchuk
:star::star::star:  jlukawska
:star::star:  Guillaume Gnaegi
:star:  Anthony Steele
:star:  Ben Bartholomew
:star:  DanHarltey
:star:  Fabrizio Mancin
:star:  Freddy
:star:  Kevin Grossmann
:star:  Marco, Roman, Thiago Loureiro, Vijay Karavadra, guoyongchang, leonluc-dev, raman-m, zhangq

---
1<sup>st</sup> :1st_place_medal: goes to **DanHarltey** for delivering **1** feature in **8** files changed 
2<sup>nd</sup> :2nd_place_medal: goes to **Ben Bartholomew** for delivering **1** feature in **3** files changed with **83** insertions 
3<sup>rd</sup> :3rd_place_medal: goes to **Guillaume Gnaegi** for delivering **1** feature in **3** files changed with **13** insertions and **422** deletions

**Comment by raman-m** [comment]
### Features Included in the Release
<sub>6aa94b54 by **raman-m** on Tuesday, October 03 at 11:05 &rarr;</sub>
Rename Release Notes file
<sub>e5f31ef0 by **Raman Maksimchuk** on Monday, October 02 at 19:17 &rarr;</sub>
Follow up #1670: Fix StyleCop issues and improve the code (#1711)
<sub>e950cf2f by **Ben Bartholomew** on Saturday, September 30 at 21:50 &rarr;</sub>
#653 Adding null check for downstreamResponse (#1334)
<sub>5a81cce1 by **Freddy** on Saturday, September 30 at 20:34 &rarr;</sub>
Second parameter of Substring is wrong in HeaderFindAndReplaceCreator (#1424)
<sub>5dbbbefc by **Fabrizio Mancin** on Saturday, September 30 at 17:26 &rarr;</sub>
#1400 Manage multiple patterns for allowed/blocked IPs via Security Options config section (#1399)
<sub>cc0b9b83 by **Kevin Grossmann** on Friday, September 29 at 21:42 &rarr;</sub>
Ignore port when using UpstreamHost option (#1505)
<sub>f8eb5e1c by **Guillaume Gnaegi** on Friday, September 29 at 20:44 &rarr;</sub>
#1634 #1487 #1329 #1304 #1294 #793 Consul polling of services: enhancements and fix errors (#1670)
<sub>ab294423 by **Roman** on Friday, September 29 at 11:27 &rarr;</sub>
#1509 #1683 Replace non-WS protocols for the 'ClientWebSocket' in WebSocketsProxyMiddleware (#1689)
<sub>190b0017 by **leonluc-dev** on Thursday, September 28 at 19:14 &rarr;</sub>
#1225 Update ServiceDiscovery documentation and samples to include Custom Providers (#1656)
<sub>fa179bf4 by **zhangq** on Thursday, September 28 at 18:04 &rarr;</sub>
#1375 #1237 #925 #920 Fix DownstreamRoute DangerousAcceptAnyServerCertificateValidator (#1377)
<sub>5fd5bf93 by **Raman Maksimchuk** on Monday, September 25 at 18:00 &rarr;</sub>
#1681 'using' statements life hacks (#1682)
<sub>8b892574 by **Raman Maksimchuk** on Monday, September 25 at 16:54 &rarr;</sub>
#1700 Add sequential tests with disabled parallelization (#1705)
<sub>fdad15d4 by **Guillaume Gnaegi** on Monday, September 25 at 15:15 &rarr;</sub>
#1700 Create custom Ocelot config file when instantiating steps during parallel execution (#1703)
<sub>75cd0b30 by **Raman Maksimchuk** on Monday, September 25 at 10:51 &rarr;</sub>
#1677 #1686 United documentation update (#1678)
<sub>b7976fcc by **guoyongchang** on Saturday, September 23 at 17:30 &rarr;</sub>
#1351 #1510 New AddOcelot extensions to support custom ASP.NET MVC pipeline building via the IMvcCoreBuilder interface (#1655)
<sub>4d245ec7 by **Vijay Karavadra** on Saturday, September 23 at 16:49 &rarr;</sub>
#1228 #1235 #1247 Overloaded AddOcelot method for dynamic FileConfiguration construction (#1569)
<sub>5ad807d7 by **jlukawska** on Saturday, September 23 at 12:12 &rarr;</sub>
#893 Problems to cancel a request (CancellationToken) (#1367)
<sub>b27761f4 by **jlukawska** on Saturday, September 23 at 11:43 &rarr;</sub>
#1057 #1068 Add service discovery logs (#1366)
<sub>ec85b132 by **jlukawska** on Friday, September 22 at 22:39 &rarr;</sub>
#1288 Query string char missing (#1335)
<sub>8893f049 by **Marco** on Friday, September 22 at 22:25 &rarr;</sub>
fix: minor grammar fix in default rate limit response message (#1639)
<sub>e6081cef by **Thiago Loureiro** on Friday, September 22 at 22:15 &rarr;</sub>
Fixed logger message typos in the OutputCacheMiddleware class (#1315)
<sub>21bc4a4a by **DanHarltey** on Friday, September 22 at 22:05 &rarr;</sub>
#908 Quality of Service - Configuration (#1279)
<sub>455d7d3f by **Anthony Steele** on Friday, September 22 at 21:45 &rarr;</sub>
Exception mapper naming (#1211)

**Comment by raman-m** [comment]
## September 2023 (version 19.1.0)


### Honoring :medal_sports: aka Top Contributors :clap:
1<sup>st</sup> :1st_place_medal: goes to **raman-m** for delivering **7** features 
2<sup>nd</sup> :2nd_place_medal: goes to **Raman Maksimchuk** for delivering **4** features 
3<sup>rd</sup> :3rd_place_medal: goes to **jlukawska** for delivering **3** features 

### Starring :star: aka Release Influencers :bowtie:
:star::star::star::star::star::star::star:  raman-m
:star::star::star::star:  Raman Maksimchuk
:star::star::star:  jlukawska
:star::star:  Guillaume Gnaegi
:star:  Anthony Steele
:star:  Ben Bartholomew
:star:  DanHarltey
:star:  Fabrizio Mancin
:star:  Freddy
:star:  Kevin Grossmann
:star:  Marco
:star:  Roman
:star:  Thiago Loureiro
:star:  Vijay Karavadra
:star:  guoyongchang
:star:  leonluc-dev
:star:  zhangq

### Features Included in the Release
<sub>0397e1b1 by **raman-m** on Thursday, October 05 at 12:31 &rarr;</sub>
Re-add release notes file
<sub>977a1bae by **raman-m** on Thursday, October 05 at 12:26 &rarr;</sub>
Fix bug in parsing of output of "git shortlog" command
<sub>e8d7edce by **raman-m** on Thursday, October 05 at 12:24 &rarr;</sub>
Delete ReleaseNotes.md
<sub>8fdc654a by **raman-m** on Wednesday, October 04 at 22:35 &rarr;</sub>
Add () after type  :laughing:
<sub>2ead70a3 by **raman-m** on Wednesday, October 04 at 22:21 &rarr;</sub>
Use Func delegate instead of static local function
<sub>997f907f by **raman-m** on Wednesday, October 04 at 21:58 &rarr;</sub>
Refactor the CreateReleaseNotes task
<sub>3485d1a5 by **raman-m** on Tuesday, October 03 at 11:05 &rarr;</sub>
Rename Release Notes file
<sub>e5f31ef0 by **Raman Maksimchuk** on Monday, October 02 at 19:17 &rarr;</sub>
Follow up #1670: Fix StyleCop issues and improve the code (#1711)
<sub>e950cf2f by **Ben Bartholomew** on Saturday, September 30 at 21:50 &rarr;</sub>
#653 Adding null check for downstreamResponse (#1334)
<sub>5a81cce1 by **Freddy** on Saturday, September 30 at 20:34 &rarr;</sub>
Second parameter of Substring is wrong in HeaderFindAndReplaceCreator (#1424)
<sub>5dbbbefc by **Fabrizio Mancin** on Saturday, September 30 at 17:26 &rarr;</sub>
#1400 Manage multiple patterns for allowed/blocked IPs via Security Options config section (#1399)
<sub>cc0b9b83 by **Kevin Grossmann** on Friday, September 29 at 21:42 &rarr;</sub>
Ignore port when using UpstreamHost option (#1505)
<sub>f8eb5e1c by **Guillaume Gnaegi** on Friday, September 29 at 20:44 &rarr;</sub>
#1634 #1487 #1329 #1304 #1294 #793 Consul polling of services: enhancements and fix errors (#1670)
<sub>ab294423 by **Roman** on Friday, September 29 at 11:27 &rarr;</sub>
#1509 #1683 Replace non-WS protocols for the 'ClientWebSocket' in WebSocketsProxyMiddleware (#1689)
<sub>190b0017 by **leonluc-dev** on Thursday, September 28 at 19:14 &rarr;</sub>
#1225 Update ServiceDiscovery documentation and samples to include Custom Providers (#1656)
<sub>fa179bf4 by **zhangq** on Thursday, September 28 at 18:04 &rarr;</sub>
#1375 #1237 #925 #920 Fix DownstreamRoute DangerousAcceptAnyServerCertificateValidator (#1377)
<sub>5fd5bf93 by **Raman Maksimchuk** on Monday, September 25 at 18:00 &rarr;</sub>
#1681 'using' statements life hacks (#1682)
<sub>8b892574 by **Raman Maksimchuk** on Monday, September 25 at 16:54 &rarr;</sub>
#1700 Add sequential tests with disabled parallelization (#1705)
<sub>fdad15d4 by **Guillaume Gnaegi** on Monday, September 25 at 15:15 &rarr;</sub>
#1700 Create custom Ocelot config file when instantiating steps during parallel execution (#1703)
<sub>75cd0b30 by **Raman Maksimchuk** on Monday, September 25 at 10:51 &rarr;</sub>
#1677 #1686 United documentation update (#1678)
<sub>b7976fcc by **guoyongchang** on Saturday, September 23 at 17:30 &rarr;</sub>
#1351 #1510 New AddOcelot extensions to support custom ASP.NET MVC pipeline building via the IMvcCoreBuilder interface (#1655)
<sub>4d245ec7 by **Vijay Karavadra** on Saturday, September 23 at 16:49 &rarr;</sub>
#1228 #1235 #1247 Overloaded AddOcelot method for dynamic FileConfiguration construction (#1569)
<sub>5ad807d7 by **jlukawska** on Saturday, September 23 at 12:12 &rarr;</sub>
#893 Problems to cancel a request (CancellationToken) (#1367)
<sub>b27761f4 by **jlukawska** on Saturday, September 23 at 11:43 &rarr;</sub>
#1057 #1068 Add service discovery logs (#1366)
<sub>ec85b132 by **jlukawska** on Friday, September 22 at 22:39 &rarr;</sub>
#1288 Query string char missing (#1335)
<sub>8893f049 by **Marco** on Friday, September 22 at 22:25 &rarr;</sub>
fix: minor grammar fix in default rate limit response message (#1639)
<sub>e6081cef by **Thiago Loureiro** on Friday, September 22 at 22:15 &rarr;</sub>
Fixed logger message typos in the OutputCacheMiddleware class (#1315)
<sub>21bc4a4a by **DanHarltey** on Friday, September 22 at 22:05 &rarr;</sub>
#908 Quality of Service - Configuration (#1279)
<sub>455d7d3f by **Anthony Steele** on Friday, September 22 at 21:45 &rarr;</sub>
Exception mapper naming (#1211)

**Pull request merged**
